lib/widgets/expandable_text_widget.dart

```diff
@@ -0,0 +1,152 @@
+    required this.maxLinesToShow,
+  });
+
+  @override
+  State<ExpandableTextWidget> createState() => _ExpandableTextWidgetState();
+}
+
+class _ExpandableTextWidgetState extends State<ExpandableTextWidget> {
+  final ValueNotifier<bool> _expanded = ValueNotifier(false);
+
+  @override
+  Widget build(BuildContext context) {
+
+    final TextSpan textSpan = TextSpan(
+      text: widget.content,
+      style: context.textTheme.smallText
+    );
+
+    final TextPainter textPainter = TextPainter(
+      text: textSpan,
+      maxLines: _expanded.value ? null : widget.maxLinesToShow,
+      textDirection: TextDirection.ltr,
+    );
+
+    textPainter.layout(maxWidth: MediaQuery.of(context).size.width);
+
+    final int numberOfLines = textPainter.computeLineMetrics().length;
+    return ValueListenableBuilder(
+      valueListenable: _expanded,
+      builder: (context, values, _) {
+        return Column(
+          crossAxisAlignment: CrossAxisAlignment.start,
+          children: [
+            Container(
+              padding: const EdgeInsets.all(16),
+              decoration: BoxDecoration(
+                color: context.colorTheme.greyLight,
+                borderRadius: AppUiConfig.borderRadious,
+              ),
+              child: LayoutBuilder(
+                builder: (BuildContext context, BoxConstraints constraints) {
+                  if (!_expanded.value && numberOfLines >= widget.maxLinesToShow) {
+                    return Column(
+                      crossAxisAlignment: CrossAxisAlignment.start,
+                      children: [
+                        Text(
+                          widget.content,
+                          maxLines: widget.maxLinesToShow,
+                          overflow: TextOverflow.ellipsis,
+                          style: context.textTheme.smallText,
+                        ),
+                        SeeMoreLessWidget(
+                          textData: context.localize.see_more,
+                          isSeeMore: true,
+                          onSeeMoreLessTap: () {
+                            _expanded.value = true;
+                          },
+                        ),
+                      ],
+                    );
+                  } else {
+                    return Column(
+                      crossAxisAlignment: CrossAxisAlignment.start,
+                      children: [
+                        Text(
+                          widget.content,
+                          style: context.textTheme.smallText,
+                        ),
+                        if (_expanded.value && numberOfLines >= widget.maxLinesToShow)
+                          SeeMoreLessWidget(
+                            textData: context.localize.see_less,
+                            isSeeMore: false,
+                            onSeeMoreLessTap: () {
+                              _expanded.value = false;
+                            },
+                          ),
```

Comment:
Personally I'd reduce some dupilcated code here. I'd consider moving logic to only one widget and in there determine `SeeMoreLessWidget` 's state, max lines to show etc.  In my opinion that solution is more readable and "clean" :-)